chia/cmds/keys_funcs.py

```diff
@@ -107,28 +113,501 @@ def delete(fingerprint: int):
+    # Arbitrary HD path was specified
+    elif derive_from_hd_path is not None:
+        derivation_root_sk, hd_path_root = derive_sk_from_hd_path(master_sk, derive_from_hd_path)
+
+    # Derive child keys from derivation_root_sk
+    if derivation_root_sk is not None and hd_path_root is not None:
+        for i in range(index, index + count):
+            if hardened_derivation:
+                sk = _derive_path(derivation_root_sk, [i])
+            else:
+                sk = _derive_path_unhardened(derivation_root_sk, [i])
+            hd_path: str = (
+                " (" + hd_path_root + str(i) + ("h" if hardened_derivation else "") + ")" if show_hd_path else ""
+            )
+            key_type_str: Optional[str]
+
+            if key_type is not None:
+                key_type_str = key_type.capitalize()
+            else:
+                key_type_str = "Hardened" if hardened_derivation else "Unhardened"
+
+            print(f"{key_type_str} public key {i}{hd_path}: {sk.get_g1()}")
+            if show_private_keys:
+                print(f"{key_type_str} private key {i}{hd_path}: {private_key_string_repr(sk)}")
+
+
+@unlocks_keyring(use_passphrase_cache=True)
+def private_key_for_fingerprint(fingerprint: int) -> Optional[PrivateKey]:
+    private_keys = Keychain().get_all_private_keys()
+
+    for sk, _ in private_keys:
+        if sk.get_g1().get_fingerprint() == fingerprint:
+            return sk
+    return None
+
+
+def get_private_key_with_fingerprint_or_prompt(fingerprint: Optional[int]):
+    """
+    Get a private key with the specified fingerprint. If fingerprint is not
+    specified, prompt the user to select a key.
+    """
+
+    # Return the private key matching the specified fingerprint
+    if fingerprint is not None:
+        return private_key_for_fingerprint(fingerprint)
+
+    fingerprints: List[int] = [pk.get_fingerprint() for pk in Keychain().get_all_public_keys()]
+    while True:
+        print("Choose key:")
+        for i, fp in enumerate(fingerprints):
+            print(f"{i+1}) {fp}")
+        val = None
+        while val is None:
+            val = input("Enter a number to pick or q to quit: ")
+            if val == "q":
+                return None
+            if not val.isdigit():
+                val = None
+            else:
+                index = int(val) - 1
+                if index >= len(fingerprints):
+                    print("Invalid value")
+                    val = None
+                    continue
+                else:
+                    return private_key_for_fingerprint(fingerprints[index])
+
+
+def private_key_from_mnemonic_seed_file(filename: Path) -> PrivateKey:
+    """
+    Create a private key from a mnemonic seed file.
+    """
+
+    mnemonic = filename.read_text().rstrip()
+    seed = mnemonic_to_seed(mnemonic, "")
+    return AugSchemeMPL.key_gen(seed)
+
+
+def resolve_derivation_master_key(fingerprint: Optional[int], filename: Optional[str]) -> PrivateKey:
```

Comment:
I think what you really want here is a `Union` type.

`fingerprint_or_filename: Union[int, str]`

and then just use `if isinstance(fingerprint_or_filename, int):` etc.

Reply:
Sure I can make that change. Passing `None` for both params actually is a handled scenario -- the user will be prompted to select a fingerprint from the available keys. I can check `isinstance(fingerprint_or_filename, str)` to catch the filename case, which is the primary condition I want to check.